src/default-config-backend.cpp

```diff
@@ -34,37 +77,26 @@ static int handle_config_updated(int fd, uint32_t mask, void *data)
     }
 
     LOGD("Reloading configuration file");
-
     char buf[INOT_BUF_SIZE] __attribute__((aligned(alignof(inotify_event))));
 
     bool should_reload = false;
-    inotify_event *event;
-
-    while (true)
+    if (read_event(fd, buf))
     {
-        auto len = read(fd, buf, INOT_BUF_SIZE);
-        if (len < 0)
+        auto event = reinterpret_cast<inotify_event*>(buf);
+        if (event->len == 0)
         {
-            break;
-        }
-
-        for (char *ptr = buf; ptr < (buf + len);
-             ptr += sizeof(inotify_event) + event->len)
+            // is file, probably the config file itself
+            should_reload = true;
+        } else if (config_dir == event->name)
```

Comment:
Ah yes, a good catch. Will fix it.

Reply:
I don't know how you're testing but I'd think we should:

1) Add an IN_CREATE watch on the directory in case the file is moved and moved back, or deleted and recreated
2) If a file was created in the directory and it's our file, we should add an IN_MODIFY watch for the newly created file

It seems like we should only have to add the directory watch once, but the file watch should be added when the file exists (or has been created).

Reply:
> Ah yes, a good catch. Will fix it.

bump